public/app/plugins/panel/alertlist/AlertInstances.tsx

```diff
@@ -31,12 +32,24 @@ export const AlertInstances: FC<Props> = ({ alerts, options }) => {
     [alerts, options]
   );
 
+  const hiddenInstances = alerts.length - filteredAlerts.length;
+
+  const uncollapsible = filteredAlerts.length > 0;
+  const toggleShowInstances = uncollapsible ? toggleDisplayInstances : noop;
+
+  useEffect(() => {
+    if (filteredAlerts.length === 0) {
+      setDisplayInstances(false);
+    }
+  }, [filteredAlerts]);
+
   return (
     <div>
       {options.groupMode === GroupMode.Default && (
-        <div className={styles.instance} onClick={() => toggleDisplayInstances()}>
-          <Icon name={displayInstances ? 'angle-down' : 'angle-right'} size={'md'} />
+        <div className={uncollapsible ? styles.clickable : ''} onClick={() => toggleShowInstances()}>
+          {uncollapsible && <Icon name={displayInstances ? 'angle-down' : 'angle-right'} size={'md'} />}
           <span>{`${filteredAlerts.length} ${pluralize('instance', filteredAlerts.length)}`}</span>
+          {hiddenInstances > 0 && <span>, {`${hiddenInstances} not shown`}</span>}
```

Comment:
Should we provide a bit more clear message like `X filtered out`? I think it would be good to emphasize that some instances are hidden due to the user's configuration

Reply:
Good point, I'll try to figure out how to clarify it without being too verbose 👍